pkg/platform/kube/functionres/lazy.go

```diff
@@ -2451,40 +2456,40 @@ func (lc *lazyClient) isPodAutoScaledUp(ctx context.Context, pod v1.Pod) (bool,
 	return false, nil
 }
 
-func (lc *lazyClient) populateDefaultContainerResources(ctx context.Context, container *v1.Container) {
+func (lc *lazyClient) populateDefaultContainerResources(ctx context.Context, resources *v1.ResourceRequirements) {
 
 	defaultFunctionPodResources := lc.platformConfigurationProvider.GetPlatformConfiguration().Kube.DefaultFunctionPodResources
 
 	lc.logger.DebugWithCtx(ctx,
-		"Populating container resources with default values",
+		"Populating resources with default values",
 		"defaultFunctionPodResources", defaultFunctionPodResources)
 
-	if container.Resources.Requests == nil {
-		container.Resources.Requests = make(v1.ResourceList)
+	if resources.Requests == nil {
+		resources.Requests = make(v1.ResourceList)
 
-		container.Resources.Requests["cpu"] = common.ParseQuantityOrDefault(defaultFunctionPodResources.Requests.CPU,
+		resources.Requests["cpu"] = common.ParseQuantityOrDefault(defaultFunctionPodResources.Requests.CPU,
 			"25m",
 			lc.logger)
-		container.Resources.Requests["memory"] = common.ParseQuantityOrDefault(defaultFunctionPodResources.Requests.Memory,
+		resources.Requests["memory"] = common.ParseQuantityOrDefault(defaultFunctionPodResources.Requests.Memory,
 			"1Mi",
 			lc.logger)
 	}
-	if container.Resources.Limits == nil {
-		container.Resources.Limits = make(v1.ResourceList)
+	if resources.Limits == nil {
+		resources.Limits = make(v1.ResourceList)
 
 		cpuQuantity, err := apiresource.ParseQuantity(defaultFunctionPodResources.Limits.CPU)
 		if err == nil {
-			container.Resources.Limits["cpu"] = cpuQuantity
+			resources.Limits["cpu"] = cpuQuantity
 		}
 		memoryQuantity, err := apiresource.ParseQuantity(defaultFunctionPodResources.Limits.Memory)
 		if err == nil {
-			container.Resources.Limits["memory"] = memoryQuantity
+			resources.Limits["memory"] = memoryQuantity
```

Comment:
If someone is putting a value to the CPU requests, but leaving the memory requests empty, it doesn't mean we don't want to enrich it from the defaults, IIUC the logic here won't do that
In other words, each field should be examined separately to be enriched with the defaults if the value is not set